packages/client/hmi-client/src/model-representation/petrinet/petrinet-service.ts

```diff
@@ -188,9 +185,6 @@ export const addState = (amr: Model, id: string, name: string) => {
 };
 
 export const addTransition = (amr: Model, id: string, name: string, value?: number) => {
-	if (amr.model.transitions.find((t) => t.id === id)) {
-		return;
-	}
```

Comment:
Why are we changing "addState" and "addTransition" logic? We don't want to allow states or transitions with the same IDs

Reply:
Yes, I originally added this condition for that reason, however I thought that it might be confusing that a function named `addTransition` may or may not actually add a transition, and that perhaps we should leave the checking of whether the id is a duplicate up to the invoker of the function. Thoughts?

Reply:
Okay, can you double check the invokers are doing the right things then?

Also in the future it might be better to do cleaning-up/refactoring as a separate PR, it is hard to follow when there are feature engineering and other things going on at the same time.